lib/srv/discovery/discovery.go

```diff
@@ -1327,8 +1322,78 @@ func (s *Server) getAllGCPServerFetchers() []server.Fetcher {
 	return allFetchers
 }
 
+// acquireDiscoveryGroup tries to acquire a lock if the Service has a DiscoveryGroup
+// It will retry using an exponential backoff algorithm.
+func (s *Server) acquireDiscoveryGroup() error {
```

Comment:
Do we need to bother with also canceling all in-progress operations with a context, or is this mostly to avoid doing the same work twice?